pkg/agent/controller/noderoute/node_route_controller.go

```diff
@@ -484,41 +486,53 @@ func (c *Controller) addNodeRoute(nodeName string, node *v1.Node) error {
 }
 
 // createIPSecTunnelPort creates an IPSec tunnel port for the remote Node if the
-// tunnel does not exist.
-func (c *Controller) createIPSecTunnelPort(nodeName string, nodeIP net.IP) error {
+// tunnel does not exist, and returns the ofport number.
+func (c *Controller) createIPSecTunnelPort(nodeName string, nodeIP net.IP) (int32, error) {
 	interfaceConfig, ok := c.interfaceStore.GetNodeTunnelInterface(nodeName)
 	if ok {
 		// TODO: check if Node IP, PSK, or tunnel type changes. This can
 		// happen if removeStaleTunnelPorts fails to remove a "stale"
 		// tunnel port for which the configuration has changed.
-		return nil
+		if interfaceConfig.OFPort != 0 {
+			return interfaceConfig.OFPort, nil
+		}
+	} else {
+		portName := util.GenerateNodeTunnelInterfaceName(nodeName)
+		ovsExternalIDs := map[string]interface{}{ovsExternalIDNodeName: nodeName}
+		portUUID, err := c.ovsBridgeClient.CreateTunnelPortExt(
+			portName,
+			c.networkConfig.TunnelType,
+			0, // ofPortRequest - let OVS allocate OFPort number.
+			nodeIP.String(),
+			c.networkConfig.IPSecPSK,
+			ovsExternalIDs)
+		if err != nil {
+			klog.Errorf("Failed to create OVS IPSec tunnel port for Node %s: %v", nodeName, err)
```

Comment:
I may be wrong but it seems this error will get logged twice: once here and once in `syncNodeRoute`.